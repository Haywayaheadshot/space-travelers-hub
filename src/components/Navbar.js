import React from 'react';
import { NavLink } from 'react-router-dom';
import '../styles/navbar.css';
import logo from './images/planet.png';

function Navbar() {
  return (
    <header className="nav-div">
      <section className="logo-div">
        <img className="space-hub-logo" src={logo} alt="Space Hub Logo" />
        <h1 className="logo">Space Travelers&apos; Hub</h1>
      </section>
      <nav className="nav-links">
        <NavLink to="/" activeclassname="active-link">
          Rockets
        </NavLink>
        <NavLink to="/missions" activeclassname="active-link">
          Missions
        </NavLink>
        <NavLink to="/profile" activeclassname="active-link">
          My Profile
        </NavLink>
      </nav>
    </header>
  );
}

export default Navbar;
