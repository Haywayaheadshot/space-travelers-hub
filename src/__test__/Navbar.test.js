import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../components/Navbar';

test('Render Navbar', () => {
  const renderNavbar = renderer.create(<BrowserRouter><Navbar /></BrowserRouter>).toJSON();
  expect(renderNavbar).toMatchSnapshot();
});
