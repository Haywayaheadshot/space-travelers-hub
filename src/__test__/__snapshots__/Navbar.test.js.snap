// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navbar 1`] = `
<nav
  className="nav-div"
>
  <section
    className="logo-div"
  >
    <img
      alt="Space Hub Logo"
      className="space-hub-logo"
      src="planet.png"
    />
    <h1
      className="logo"
    >
      Space Travelers' Hub
    </h1>
  </section>
  <section
    className="hamburger-div for-phone"
  >
    <div
      aria-expanded={false}
      aria-label="Show menu"
      className="hamburger-react"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      style={
        Object {
          "cursor": "pointer",
          "height": "48px",
          "outline": "none",
          "position": "relative",
          "transform": "none",
          "transition": "0.8s cubic-bezier(0, 0, 0, 1)",
          "userSelect": "none",
          "width": "48px",
        }
      }
      tabIndex={0}
    >
      <div
        style={
          Object {
            "background": "#0290ff",
            "height": "3px",
            "left": "8px",
            "position": "absolute",
            "top": "13px",
            "transform": "none",
            "transition": "0.8s cubic-bezier(0, 0, 0, 1)",
            "width": "32px",
          }
        }
      />
      <div
        style={
          Object {
            "background": "#0290ff",
            "height": "3px",
            "left": "8px",
            "position": "absolute",
            "top": "23px",
            "transform": "none",
            "transition": "0.8s cubic-bezier(0, 0, 0, 1)",
            "width": "32px",
          }
        }
      />
      <div
        style={
          Object {
            "background": "#0290ff",
            "height": "3px",
            "left": "8px",
            "position": "absolute",
            "top": "33px",
            "transform": "none",
            "transition": "0.8s cubic-bezier(0, 0, 0, 1)",
            "width": "32px",
          }
        }
      />
    </div>
  </section>
  <ul
    className="nav-links for-desktop"
  >
    <li
      className="nav-desk-li"
    >
      <a
        activeclassname="active-link"
        aria-current="page"
        className="links active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li
      className="nav-desk-li"
    >
      <a
        activeclassname="active-link"
        className="links"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li
      className="nav-desk-li"
    >
      <a
        activeclassname="active-link"
        className="links"
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
